src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -1498,7 +1504,10 @@ private void login(String primary,
         // for all other cases, including multiSubnetFailover
         final boolean isDBMirroring = (null == mirror && null == foActual) ? false : true;
         int sleepInterval = 100;  // milliseconds to sleep (back off) between attempts.
-        long timeoutUnitInterval;
+        long timeoutUnitInterval = 0;
```

Comment:
These additional variable allocations can be avoided. You can enhance the logic by checking for `isDBMirroring` `userParallel` `useTnir` to decide whether TIMEOUTSTEP or TIMEOUTSTEP_TNIR needs to be used and continue to use intervalExpire to populate the correct time out in this context.